gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -2265,11 +2264,12 @@ void Movable::DoStep(unsigned int walkScale, ieDword time) {
 
 		if (BlocksSearchMap() && actorInTheWay && actorInTheWay != this && actorInTheWay->BlocksSearchMap()) {
 			// Give up instead of bumping if you are close to the goal
-			if (!(step->Next) && std::abs(nmptStep.x - Pos.x) < XEPS && std::abs(nmptStep.y - Pos.y) < YEPS) {
+			if (!(step->Next) && PersonalDistance(nmptStep, this) < MAX_OPERATING_DISTANCE) {
```

Comment:
Did you use PersonalDistance deliberately? It will shrink the distance for actors and considering the sizes involved, I'm surprised dragons can still move.

Reply:
I used it for consistency with WithinPersonalRange which is used by MoveNearerTo, as well as with the PersonalDistance checks in e.g. UseItem. I tested it with the draconis battle and did not find any issues, but I still have some more tests to conduct which I plan to do today.

Reply:
If you can move him around after ctrl-q, then it should be fine.

Reply:
Yup, works with very close moves too.

Reply:
Thanks, I'll merge it then.